File: app/java/src/main/java/com/example/production/domain/model/purchase/UnitPrice.java
package com.example.production.domain.model.purchase;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UnitPrice {
    private Integer id;
    private String itemCode;
    private String supplierCode;
    @Builder.Default
    private BigDecimal lotUnitQuantity = BigDecimal.ONE;
    private LocalDate effectiveFrom;
    private LocalDate effectiveTo;
    private BigDecimal unitPrice;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Purchase/UnitPrice.cs
namespace ProductionManagement.Domain.Models.Purchase;

/// <summary>
/// 単価マスタ
/// </summary>
public class UnitPrice
{
    public int Id { get; set; }
    public required string ItemCode { get; set; }
    public required string SupplierCode { get; set; }
    public decimal LotUnitQuantity { get; set; } = 1m;
    public required DateOnly EffectiveFrom { get; set; }
    public DateOnly? EffectiveTo { get; set; }
    public required decimal Price { get; set; }
    public DateTime? CreatedAt { get; set; }
    public string? CreatedBy { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string? UpdatedBy { get; set; }
}
